ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/internalresponse/EthSignTransactionResultProvider.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.requesthandler.internalresponse;
+
+import static tech.pegasys.ethsigner.core.jsonrpc.response.JsonRpcError.INVALID_PARAMS;
+import static tech.pegasys.ethsigner.core.jsonrpc.response.JsonRpcError.SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT;
+
+import tech.pegasys.ethsigner.core.AddressIndexedSignerProvider;
+import tech.pegasys.ethsigner.core.jsonrpc.EthSendTransactionJsonParameters;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonDecoder;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.core.jsonrpc.exception.JsonRpcException;
+import tech.pegasys.ethsigner.core.requesthandler.ResultProvider;
+import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction.EthTransaction;
+import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction.Transaction;
+import tech.pegasys.signers.secp256k1.api.Signature;
+import tech.pegasys.signers.secp256k1.api.Signer;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.web3j.crypto.Sign;
+import org.web3j.crypto.TransactionEncoder;
+import org.web3j.utils.Numeric;
+
+public class EthSignTransactionResultProvider implements ResultProvider<String> {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final long chainId;
+  private final AddressIndexedSignerProvider signerProvider;
+  private final JsonDecoder decoder;
+
+  public EthSignTransactionResultProvider(
+      final long chainId,
+      final AddressIndexedSignerProvider signerProvider,
+      final JsonDecoder decoder) {
+    this.chainId = chainId;
+    this.signerProvider = signerProvider;
+    this.decoder = decoder;
+  }
+
+  @Override
+  public String createResponseResult(final JsonRpcRequest request) {
+    // Signs a transaction that can be submitted to the network at a later time using with
+    // eth_sendRawTransaction.
+    LOG.debug("Transforming request {}, {}", request.getId(), request.getMethod());
+    final Transaction transaction;
+    try {
+      transaction = createTransaction(request);
```

Comment:
the Nonce is deemed an optional field - which for this operation, does not hold true.
Is it worth doing some extra checks to ensure the transaction is fully populated? Otherwise the RLP encoding is going to either crash, or be invalid.